hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -589,6 +740,11 @@ def _generate_flame_clip_name(context, publish_fields):
 
     name = ""
 
+    # If we have template fields passed in, then we'll try to extract
+    # some information from them. If we don't, then we fall back on
+    # some defaults worked out below.
+    publish_fields = publish_fields or dict()
```

Comment:
@manneohrstrom Any suggestions on how we might better handle the situation here without any template fields to make use of? I'm especially concerned about this always defaulting to version 0 when we aren't using a template. Maybe there's some data I can be pulling that publish2 has already figured out?

Reply:
when we are not using templates, can't we use the versioning coming from the fact that we are publishing all things in shotgun? So with each change to the clip xml file, we are also creating a published file in sg. I think it makes sense from a workflow perspective if the entries in the clip files reflect the version history in shotgun of the clip file. Does that make sense?